index.js

```diff
@@ -29,20 +29,21 @@ exports.sync = function (store, router) {
     }
     var to = transition.to
     currentPath = to.path
-    store.dispatch('router/ROUTE_CHANGED', to)
+    var commit = store.commit || store.dispatch
+    commit('router/ROUTE_CHANGED', to)
   })
 }
 
 function patchStore (store) {
   // add state
   var set = store._vm.constructor.set
-  store._dispatching = true
+  store._dispatching = store._committing = true
```

Comment:
ideally we want to avoid setting an extra private property on the store. Maybe check for a conditional `store.hasOwnProperty('_dispatching')`.